qbraid/runtime/azure/result_builder.py

```diff
@@ -346,3 +346,66 @@ def get_counts(self) -> Union[dict[str, int], list[dict[str, int]]]:
             return results[0]["data"]["counts"] if results[0]["success"] else {}
 
         return [result["data"]["counts"] if result["success"] else {} for result in results]
+
+
+class AzureAHSModelResultBuilder:
```

Comment:
wouldn't it be easier / require less redundant code to just add a format ahs result method to the existing result builder class? seems to be a fair amount of overlap

Reply:
That's true, but it didn't make sense since from a naming convention to add an analog hamiltonian simulation formatter in a GateModelResultBuilder.

Reply:
If you're okay with the name change to AzureResultBuilder I can do that, just wasn't sure if the convention was to separate Gate vs AHS vs Annealing etc per provider similar to Braket.